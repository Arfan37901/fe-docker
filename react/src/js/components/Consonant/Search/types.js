import {
    bool,
    func,
    string,
} from 'prop-types';

export const SearchType = {
    name: string,
    value: string,
    autofocus: bool,
    placeholderText: string,
    onSearch: func.isRequired,
};

export const SearchIconType = {
    onClick: func.isRequired,
};
