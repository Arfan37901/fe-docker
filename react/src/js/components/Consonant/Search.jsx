import React from 'react';
import PropTypes from 'prop-types';

const Search = (props) => {
    const { itemsQty, value, onSearch } = props;
    const handleSearch = (evt) => {
        onSearch(evt.target.value);
    };

    const clearSearch = () => {
        onSearch('');
    };

    return (
        <div className="consonant-search">
            <label>
                <span className="consonant-search--input-title">Search</span>
                <span className="consonant-search--input-wrapper">
                    <input
                        type="search"
                        placeholder={onSearch ? `Search within ${itemsQty} items` : 'Search'}
                        value={value}
                        onChange={handleSearch}
                        required />
                    <button
                        type="button"
                        title="Click to clear search query"
                        className="consonant-search--input-clear"
                        onClick={clearSearch}>clear
                    </button>
                </span>
            </label>
        </div>
    );
};

export default Search;

Search.propTypes = {
    itemsQty: PropTypes.number,
    onSearch: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired,
};

Search.defaultProps = {
    itemsQty: 0,
};
