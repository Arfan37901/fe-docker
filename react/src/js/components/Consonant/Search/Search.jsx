import PropTypes from 'prop-types';
import React from 'react';

const searchId = 'consonant-search';
const Search = (props) => {
    const {
        value,
        onSearch,
        placeholderText,
        leftPanelTitle,
        name,
        autofocus,
    } = props;

    let textInput = null;
    const handleSearch = (evt) => {
        onSearch(evt.target.value);
    };
    const setAutofocus = () => {
        textInput.focus();
    };
    const clearSearch = () => {
        onSearch('');
        setAutofocus();
    };
    return (
        <div data-testid={name} className="consonant-search">
            <label htmlFor={searchId}>
                <span className="consonant-search--input-title">{leftPanelTitle}</span>
                <span className="consonant-search--input-wrapper">
                    <input
                        id={searchId}
                        data-testid="search-input"
                        type="search"
                        placeholder={placeholderText}
                        onClick={e => e.stopPropagation()}
                        value={value}
                        onChange={handleSearch}
                        ref={(input) => {
                            if (autofocus && input) input.focus();
                            textInput = input;
                        }}
                        className="consonant-search--input"
                        required />
                    <button
                        type="button"
                        title="Click to clear search query"
                        className="consonant-search--input-clear"
                        onClick={clearSearch}
                        tabIndex="0">clear
                    </button>
                </span>
            </label>
        </div>
    );
};

export default Search;

Search.propTypes = {
    name: PropTypes.string,
    onSearch: PropTypes.func.isRequired,
    value: PropTypes.string,
    placeholderText: PropTypes.string,
    leftPanelTitle: PropTypes.string,
    autofocus: PropTypes.bool,
};

Search.defaultProps = {
    value: '',
    name: 'consonant-search',
    placeholderText: 'Search here...',
    leftPanelTitle: 'Search',
    autofocus: true,
};
