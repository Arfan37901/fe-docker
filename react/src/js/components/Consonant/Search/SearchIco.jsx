import React from 'react';
import PropTypes from 'prop-types';

const SearchIco = (props) => {
    const { onClick } = props;

    return (
        <button
            type="button"
            className="search-ico"
            onClick={onClick}
            tabIndex="0">
            Click to search
            <span />
        </button>
    );
};

export default SearchIco;

SearchIco.propTypes = {
    onClick: PropTypes.func.isRequired,
};
