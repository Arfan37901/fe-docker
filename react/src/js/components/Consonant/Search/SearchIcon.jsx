import React from 'react';
import { func } from 'prop-types';

const TSearchIcon = {
    onClick: func.isRequired,
};

/**
 * Search Icon (Used in Top Filter View Only)
 *
 * Only function it has is to toggle Search Box
 *
 * @component
 * @example
 * const props= {
    onClick: Function,
 * }
 * return (
 *   <Select {...props}/>
 * )
 */
const SearchIcon = (props) => {
    const { onClick } = props;

    return (
        <button
            data-testid="search-icon"
            type="button"
            className="search-ico"
            onClick={onClick}
            tabIndex="0">
            <span />
        </button>
    );
};

SearchIcon.propTypes = TSearchIcon;

export default SearchIcon;
