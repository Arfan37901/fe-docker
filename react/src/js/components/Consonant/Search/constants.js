export const SearchInputDefaultProps = {
    name: '',
    value: '',
    autofocus: true,
    placeholderText: '',
};

/**
 * Used as unique id for accessibility labels/attributes
 * @type {Number}
 */
export const searchInputId = 'consonant-search';
