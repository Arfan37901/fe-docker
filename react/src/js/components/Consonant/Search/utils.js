// eslint-disable-next-line import/prefer-default-export
export const selector = ({
    search: {
        i18n: {
            leftFilterPanel: {
                searchTitle,
            },
        },
    },
}) => ({
    leftPanelTitle: searchTitle,
});
