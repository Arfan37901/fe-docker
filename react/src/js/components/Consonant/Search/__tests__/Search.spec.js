import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Search from '../Search';
import setup from '../../Helpers/Testing/Utils/Settings';
import { DEFAULT_PROPS } from '../../Helpers/Testing/Constants/Search';

const renderSearch = setup(Search, DEFAULT_PROPS);

describe('Search Component', () => {

    test('Should be able to handle focus and blur events', () => {
        renderSearch({ onBlur: null });

        const inputElement = screen.getByTestId('search-input');

        fireEvent.focus(inputElement);
        fireEvent.blur(inputElement);
    });
    test('Should be able to handle searches', () => {
        const { props: { onSearch } } = renderSearch({ value: 'First Search Query' });

        const inputElement = screen.getByTestId('search-input');

        expect(inputElement.value).toBe('First Search Query');

        fireEvent.change(inputElement, { target: { value: 'Second Search Query' } });

        expect(onSearch).toBeCalled();
        expect(onSearch).toBeCalledWith('Second Search Query');

        onSearch.mockClear();
    });
    test('Should be able to clear search values', () => {
        const { props: { onSearch } } = renderSearch();

        const buttonElement = screen.queryByTestId('clear-search-button');

        fireEvent.click(buttonElement);

        expect(onSearch).toBeCalled();
        expect(onSearch).toBeCalledWith('');
    });
});
