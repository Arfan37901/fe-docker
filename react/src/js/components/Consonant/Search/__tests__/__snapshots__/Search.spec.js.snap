// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Search Check snapshots should renders correctly with value 1`] = `
<div
  className="consonant-search"
  data-testid="consonant-search"
>
  <label
    htmlFor="consonant-search"
  >
    <span
      className="consonant-search--input-title"
    >
      Search
    </span>
    <span
      className="consonant-search--input-wrapper"
    >
      <input
        className="consonant-search--input"
        data-testid="search-input"
        id="consonant-search"
        onChange={[Function]}
        placeholder="search text"
        required={true}
        type="search"
        value=""
      />
      <button
        className="consonant-search--input-clear"
        onClick={[Function]}
        tabIndex="0"
        title="Click to clear search query"
        type="button"
      >
        clear
      </button>
    </span>
  </label>
</div>
`;

exports[`Consonant/Search Check snapshots should renders correctly without value 1`] = `
<div
  className="consonant-search"
  data-testid="consonant-search"
>
  <label
    htmlFor="consonant-search"
  >
    <span
      className="consonant-search--input-title"
    >
      Search
    </span>
    <span
      className="consonant-search--input-wrapper"
    >
      <input
        className="consonant-search--input"
        data-testid="search-input"
        id="consonant-search"
        onChange={[Function]}
        placeholder="search text"
        required={true}
        type="search"
        value="search value"
      />
      <button
        className="consonant-search--input-clear"
        onClick={[Function]}
        tabIndex="0"
        title="Click to clear search query"
        type="button"
      >
        clear
      </button>
    </span>
  </label>
</div>
`;
